### ISSUE TITLE
StaticGeometry gets double the Root SceneNode transform

### ISSUE THREAD

**Issue by chillywillysoft** [OGRECave/ogre#2510]
#### System Information
- Ogre Version: 13.4.1
- Operating System / Platform: macOS Catalina
- RenderSystem: OpenGL

#### Detailled description
When StaticGeometry is created using StaticGeometry::addSceneNode(), each Entity is queued along with its derived transform. That bakes the Root SceneNode's transform into the geometry. Then when StaticGeometry::build() calls Region::build(), it creates its Node as a child of the Root SceneNode. This results in the Root SceneNode's transform being applied twice.

Not sure best way to fix:
1. Straightforward: In StaticGeometry::addSceneNode(), instead of calling _getDerivedPosition() (etc) we manually derive from a child of Root.
2. Tricky: Use _getDerivedPosition() (etc) but then subtract or divide away the Root's transform.
3. Ugly: Save the Root's transform, set it to identity, call _getDerivedPosition() (etc), then restore the Root's transform.

**Comment by paroj** [comment]
how about setting `Matrix4::IDENTITY` here:
https://github.com/OGRECave/ogre/blob/bc3221a5e582f8ffdad54eb35b6c015c4330a975/OgreMain/src/OgreStaticGeometry.cpp#L1368

**Comment by chillywillysoft** [comment]
I'm not sure if I understand. I'm not familiar with the getWorldTransforms() function but I'm only seeing it being called from BillboardSet and AutoParamDataSource. Would that get called while creating the StaticGeometry or while rendering it?

If while rendering, then it would ignore the Root Node's current transform and only use what was baked in. If the transform on the Root node changes after the StaticGeometry is made, it would not track with the rest of the Scene.

While creating the the StaticGeometry, StaticGeometry::addSceneNode() calls Node::_getDerivedPosition(), Node::_getDerivedOrientation(), and Node::_getDerivedScale()), all of which call _updateFromParent() and it recurses up the tree.

**Comment by chillywillysoft** [comment]
> thats kind of the premise of StaticGeometry. If any intermediate node changes, StaticGeometry would not reflect that either
Fair enough. Can we avoid applying any transforms at all while rendering or do we have to pass something to the GPU? Out of curiosity can you point me to the rendering code that calls getWorldTransforms()? I haven't been able to find it.

**Comment by paroj** [comment]
what your write from that function gets almost directly passed to the GPU - e.g. here: https://github.com/OGRECave/ogre/blob/5f06d69188d6bc39a75893388ddd0f811f6380fd/OgreMain/src/OgreGpuProgramParams.cpp#L1864

**Comment by chillywillysoft** [comment]
I see, thank you.

Returning `Matrix4::IDENTITY` from getWorldTransforms() sounds good for rendering.

What about culling? Will it still use the Root node's transform while culling?

**Comment by paroj** [comment]
> What about culling? Will it still use the Root node's transform while culling?

Yes. I dont know all the paths that culling takes off the top of my head. Overriding _notifyCurrentCamera will disable culling by the far plane altogether.
There are also aab and bounding sphere tests though. We should not just disable culling. The individual StaticGeometry Regions should still be culled. Also they need the correct camera distance for LOD AFAIR

**Comment by paroj** [comment]
given that transforms on the root scene node have no observable effect, I am inclined to just warn about this now and ignore any root scene node transform in a future release.

Did you have any particular reason to do that transform?
